backend/drm/drm.c

```diff
@@ -1629,6 +1630,26 @@ void destroy_drm_connector(struct wlr_drm_connector *conn) {
 	free(conn);
 }
 
+int wlr_drm_backend_get_client_fd(struct wlr_backend *backend) {
+	assert(backend && wlr_backend_is_drm(backend));
+
+	struct wlr_drm_backend *drm = get_drm_backend_from_backend(backend);
+	char *path = drmGetDeviceNameFromFd2(drm->fd);
+
+	int fd = open(path, O_RDWR);
+	if (fd < 0) {
+		wlr_log_errno(WLR_ERROR, "Unable to clone DRM fd for client fd.\n");
+		free(path);
+		return -EINVAL;
+	}
+	if (drmIsMaster(fd)) {
+		drmDropMaster(fd);
+	}
+	assert(!drmIsMaster(fd) && "Won't send master fd to client");
```

Comment:
This crashes the compositor if `drmDropMaster` fails.

Also, this doesn't guarantee that the returned FD will be non-master: assertions can be disabled with `ndebug`.